django/contrib/postgres/indexes.py

```diff
@@ -27,6 +27,13 @@ def deconstruct(self):
         kwargs['pages_per_range'] = self.pages_per_range
         return path, args, kwargs
 
+    def set_name_with_model(self, model):
+        try:
+            super().set_name_with_model(model)
+        except AssertionError:
```

Comment:
OK. Did you also have in mind to consider the max length to determine the length of the parts of the index name, e.g., more than 6 characters for the hash, or more than 11 characters for the table name?

So far, I just set `max_name_length = connection.ops.max_name_length()` in `BrinIndex` based on the assumption that this index will only be created on a PostgreSQL-backend, and changed  accordingly in `Index`.